src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -394,7 +400,7 @@ struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
         auto has_ignore_index = normalize_input_index(parser, info, ignore_index, ignore_index_val);
 
         bool has_weights        = (args.size() > 2);
-        instruction_ref weights = get_weights(info, args, scores_shape, class_size);
+        instruction_ref weights = get_weights(info, op_name, args, scores_shape, class_size);
```

Comment:
Performance: Weights could be statically multiplied by -1, instead of adding an additional `neg` instruction below

Reply:
Ah yeah I see that. Good catch

Looks like I can reduce this from a possible N x Cx d1 x ... x d2 x .. x dk) to (C + 1) pointwise operations, where C is the classes, N is batch and d1..dk is the other dimensions. +1 here is because outside of adding a negative on when we acquire weights I would need a change to the weight reduction case when weights are non 1 and mean reduction is selected to flip the negative back,

Reply:
Added this, but i'll need to update parser tests for the softmaxcrossentropyloss op as this effects that operator as well.

Reply:
Tests are updated. Just need to add last test cases